plugins/woocommerce/tests/e2e-pw/tests/shopper/cart-block-coupons.spec.js

```diff
@@ -247,4 +262,49 @@ test.describe( 'Cart Block Applying Coupons', () => {
 				)
 		).toBeVisible();
 	} );
+
+	test( 'prevents cart block applying coupon with usage limit', async ( {
+		page,
+		context,
+	} ) => {
+		// go to merchant and create a new order with coupon usage 1/1
+		await page.goto( 'wp-admin/post-new.php?post_type=shop_order' );
+		await page.waitForLoadState( 'networkidle' );
+		await page.locator( 'input[name="log"]' ).fill( admin.username );
+		await page.locator( 'input[name="pwd"]' ).fill( admin.password );
+		await page.locator( 'text=Log In' ).click();
+		await page.waitForLoadState( 'networkidle' );
+		page.on( 'dialog', ( dialog ) => dialog.accept( couponLimitedCode ) );
+		await page.locator( 'button.add-coupon' ).click();
+		await expect( page.locator( '.note_content' ) ).toContainText(
+			`Coupon applied: "${ couponLimitedCode }".`
+		);
+		await page.locator( 'button.save_order' ).click();
+		await expect(
+			page.locator( 'div.updated.notice.notice-success.is-dismissible', {
+				has: page.locator( 'p' ),
+			} )
+		).toContainText( 'Order updated.' );
+		await context.clearCookies(); // logs out admin
+
+		// add product to cart block and go to checkout
+		await page.goto( `/shop/?add-to-cart=${ productId }` );
+		await page.waitForLoadState( 'networkidle' );
+		await page.goto( pageSlug );
```

Comment:
I understand it, but because `pageTitle` and `pageSlug` are one aside from the other at the beginning of the test, it is kind of expected to guess `pageSlug` is related to `pageTitle`.

For page creation in the setup or env, I'd like to see that but so far no solutions found. I wanted to boost this and bring the tests anything sooner in production since we don't have any existing for block-based exp and we're prone to bugs because of that (I found already a few while working on this, e.g. #42967). We can work on it in a separate PR if you don't have any quick solution now.

Reply:
Hi @veljkho! Sorry for my late follow up, I was AFK during Christmas holidays.

>  understand it, but because pageTitle and pageSlug are one aside from the other at the beginning of the test, it is kind of expected to guess pageSlug is related to pageTitle.

Yep, that's true. It's just that we're adding additional logic (which means more complexity i.e. more execution time and more probability of introducing a bug) for something we don't need now (see the [KISS principle](https://en.wikipedia.org/wiki/KISS_principle)). 

If pageTitle had different values, then adding this logic would be totally justified as we would need a dynamic value for it. But both pageTitle and pageSlug have always the same value, so the simplest approach would be just to move them as a constant at the top of the file so that we just need to declare them once and use them in multiple tests.

Then if we realize we need multiple pageTitle and pageSlug in the future, we can of course add this logic to keep it simple and reduce maintenance. But considering this is the spec for `cart block coupons`, there is a chance that we just need to create the `cart-block` page here always, so pageTitle and pageSlug would have always the same value.

I'm trying to advocate for simplicity here because we might be adding logic that we don't need, but if you really think we do need it or can see clear scenarios where we'll need it in the near future, then I'm fine with it 👍 

> We can work on it in a separate PR if you don't have any quick solution now.
Sure thing, makes sense!
Regarding the automated creation of a page as precondition, I would suggest 2 potential approaches (I never did it this way, so some investigation for this is required, but I'm optimistic about it based on the existing documentation:
- **[WP-CLI](https://developer.wordpress.org/cli/commands/post/create/):** We might be able to create that page through WP-CLI in the `./tests/e2e-pw/bin/test-env-setup.sh` setup file OR
- **[WordPress API](https://developer.wordpress.org/rest-api/reference/pages/#create-a-page):** We could create the page directly from the test (this would have a better level of independence and the WordPress API seems to be more complete than WP CI regarding the options for creating a page).

Anyway, I agree that this could be done in a separate PR.

Thanks again, @veljkho!

Reply:
@alopezari We will anyway remove page creation inside the test soonish so these will be removed. I will however rename it to `cartBlockPage` and `cartBlockPageSlug` no worries. Just waiting for this [blocking PR](https://github.com/woocommerce/woocommerce/pull/43056) to be merged first.